File: _src/main/java/raccoonlang/parser/ast/NamespaceAstNode.java
package raccoonlang.parser.ast;

import raccoonlang.parser.Parser;
import raccoonlang.tokenizer.TokenType;

public class NamespaceAstNode {
    public FqtnAstNode namespaceName;

    public static NamespaceAstNode tryParse(Parser parser) {
        if(parser.peek().getTokenType() != TokenType.NAMESPACE){
            return null;
        }

        NamespaceAstNode namespaceAstNode = new NamespaceAstNode();

        parser.take();
        namespaceAstNode.namespaceName = FqtnAstNode.parse(parser);
        parser.take(TokenType.SEMICOLON);

        return namespaceAstNode;
    }

    @Override
    public String toString() {
        return "NamespaceAstNode{" +
                "namespaceName=" + namespaceName.toString() +
                '}';
    }
}

File: Parser/AST/NamespaceAstNode.cs
namespace Raccoonlang.Parser.AST;

using Tokenizer;

public class NamespaceAstNode
{
    public FqtnAstNode? Node {get; private set;}

    public static NamespaceAstNode? TryParse(Parser parser)
    {
        if (parser.Peek().Type != TokenType.NAMESPACE) {
            return null;
        }

        NamespaceAstNode node = new NamespaceAstNode();
        parser.Take();
        node.Node = FqtnAstNode.Parse(parser);
        parser.Take(TokenType.SEMICOLON);

        return node;
    }
}